From 9c4aed9c58cea1af9d97ca4698df3ea196bc7ed1 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:49 -0700
Subject: [PATCH 0439/1852] dt-bindings: media: xilinx: vpss-csc: Make max
 width, height mandatory

commit 67f5d488bb024ff60a2025504c1f8db4c87478a4 from
https://github.com/Xilinx/linux-xlnx.git

This patch makes the xlnx,max-width and xlnx,max-height dt properties as
mandatory. This breaks backward compatibility.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/media/xilinx/xlnx,v-gamma-lut.txt      | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-gamma-lut.txt b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-gamma-lut.txt
index f0f00a79f0e1..7bd750f009b4 100644
--- a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-gamma-lut.txt
+++ b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-gamma-lut.txt
@@ -12,6 +12,10 @@ Required properties:
 - clocks: Reference to the clock that drives the ap_clk
 	  signal of Video Gamma Lookup.
 
+- xlnx,max-height: Maximum number of lines. Valid range is 64 to 4320.
+
+- xlnx,max-width: Maximum number of pixels in a line. Valid range is 64 to 8192.
+
 - reset-gpios: Specifier for a GPIO that asserts Gamma IP (AP_RST_N) reset
 
 - ports: Video ports, using the DT bindings defined in ../video-interfaces.txt.
@@ -25,18 +29,15 @@ Required port properties:
 
 - xlnx,video-width: Video width as defined in video.txt. Can be either 8 or 10.
 
-Optional properties:
-
-- xlnx,max-height: Maximum number of lines.
-- xlnx,max-width: Maximum number of pixels in a line.
-
-Example :
+Example:
 
 	gamma_lut_1: gamma_lut_1@0xa0080000 {
 		compatible = "xlnx,v-gamma-lut";
 		reg = <0x0 0xa0080000 0x0 0x10000>;
 		clocks = <&vid_stream_clk>;
 		reset-gpios = <&gpio 83 1>;
+		xlnx,max-height = <2160>;
+		xlnx,max-width = <3840>;
 		ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.31.1

